system/jlib/jlzw.cpp

```diff
@@ -2301,15 +2303,10 @@ class CCompressedFileReader final : public CCompressedFileBase
         {
             if (!expander)
             {
-                if (compMethod == COMPRESS_METHOD_FASTLZ)
-                    expander.setown(createFastLZExpander());
-                else if (compMethod == COMPRESS_METHOD_LZ4)
-                    expander.setown(createLZ4Expander());
-                else // fallback
-                {
-                    compMethod = COMPRESS_METHOD_LZW;
-                    expander.setown(createLZWExpander(true));
-                }
+                ICompressHandler *handler = queryCompressHandler(compMethod);
+                if (unlikely(!handler))
```

Comment:
can compMethod be COMPRESS_METHOD_NONE at this point, and if so, is default used here, or will this be null and throw an exception?
Probably as intended, if NONE throws an unsupported message.

Reply:
If it was it would throw an exception - but that should never be the case, it has been extracted from the compressed file footer.